api/src/main/java/org/apache/iceberg/expressions/ManifestEvaluator.java

```diff
@@ -142,6 +142,25 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_MATCH;
     }
 
+    @Override
+    public <T> Boolean isNaN(BoundReference<T> ref) {
+      int pos = Accessors.toPosition(ref.accessor());
+      // containsNull encodes whether at least one partition value is null, lowerBound is null if
+      // all partition values are null.
+      ByteBuffer lowerBound = stats.get(pos).lowerBound();
+      if (lowerBound == null) {
```

Comment:
To be safe, I think this should validate that `containsNull` is true.

Reply:
You mean check for both `containsNull` and `stats.get(pos).lowerBound() == null` are true? When would `lowerBound` be null while the column doesn't contain null? I guess I'll also need to update `notNull` for this too (since I copied the logic from there)?

Reply:
Seems like #1803 is missing `PartitionFieldSummary.containsNaN()`, or is it in some other PR?

Reply:
That should be doable, although I originally consider the scope of the NaN support to be only on manifest entry level, I  wasn't sure if we want to extend it beyond that?

Reply:
I'm not sure of a case where it would happen, but `containsNull` is the source of truth for whether there are null values, not a missing bound value.

Reply:
Sounds good, I'll add `containsNull` to both here and `notNull`. And looks like we do want to update `PartitionFieldSummary`, that I'll do in a separate pr.